### PR TITLE
feat(core): add success notification support for auth provider methods #5473

### LINKED ISSUE #5473: [FEAT] Add `successNotification` field to AuthProvider methods' return type.
### Is your feature request related to a problem? Please describe.

Currently, if any of the Auth Provider methods returns `success: false` and `error` fields, we are showing error notification.

But it's not possible to do the same easily for success notifications.

### Describe alternatives you've considered

_No response_

### Additional context

https://github.com/refinedev/refine/discussions/5170#discussioncomment-7619752

### Describe the thing to improve

1- Add `successNotification` field to [`AuthActionResponse` type](https://github.com/refinedev/refine/blob/master/packages/core/src/interfaces/bindings/auth.ts).

```ts
type SuccessNotificationResponse {
  message: string;
  description?: string;
}

export type AuthActionResponse = {
    success: boolean;
    redirectTo?: string;
    error?: RefineError | Error;
    [key: string]: unknown;
    successNotification?: SuccessNotificationResponse
};
```

2-) Update relevant hooks to show success notification when `successNotification` object is provided and **add specs**.

- `useForgotPassword`
- `useLogin`
- `useLogout`
- `useRegister`
- `useUpdatePassword`

3-) Update relevant documentation pages.

### PR DESCRIPTION
**Pull request by Conqxeror**
#### Changes Made:

1. Added `successNotification` field to `AuthActionResponse` type.

    ```typescript
    type SuccessNotificationResponse = {
      message: string;
      description?: string;
    };

    export type AuthActionResponse = {
        success: boolean;
        redirectTo?: string;
        error?: RefineError | Error;
        [key: string]: unknown;
        successNotification?: SuccessNotificationResponse;
    };
    ```

2. Updated relevant hooks (`useForgotPassword`, `useLogin`, `useLogout`, `useRegister`, `useUpdatePassword`) to display success notifications when the `successNotification` object is provided.

### Self Check before Merge

Please check all items below before review.

-   [x] Corresponding issues are created/updated or not needed
-   [x] Docs are updated/provided or not needed
-   [x] Examples are updated/provided or not needed
-   [x] TypeScript definitions are updated/provided or not needed
-   [x] Tests are updated/provided or not needed
-   [x] Changesets are provided or not needed

### CODE REVIEW

File: packages/core/src/hooks/auth/useRegister/index.ts [hunk 1]
```diff
@@ -215,3 +220,14 @@ const buildNotification = (
...
     };
 };
+
+const buildSuccessNotification = (
+    successNotification: SuccessNotificationResponse,
+  ): OpenNotificationParams => {
+    return {
+      message: successNotification.message || "Success",
+      description: successNotification.description || "Operation completed successfully",
+      key: "success-notification",
```
Comment (BatuhanW) [inline packages/core/src/hooks/auth/useRegister/index.ts:230]:
```suggestion
      key: "register-success",
```

**Review by BatuhanW — COMMENTED** [review]
Hey @Conqxeror thanks for the PR! You can proceed with tests and docs.

Also added comment for `key` parameter. We can use unique key for each notification, like `forgot-password-success`, `login-success` etc.

Looking forward

File: packages/core/src/hooks/auth/useForgotPassword/index.spec.ts [hunk 2]
```diff
@@ -487,6 +487,45 @@ describe("useForgotPassword Hook", () => {
...
+        await waitFor(() => {
+            expect(openNotificationMock).toBeCalledWith({
+                key: "forgot-password-success",
+                type: "success",
+                message: "Password reset successful",
+                description: "Your password has been successfully reset",
+            });
+        });
+    });
+
```
Comment (arghyapolley) [inline packages/core/src/hooks/auth/useForgotPassword/index.spec.ts:528]: A well structured code but consider adding comments for explaining the behaviour of hook which will be helpful for other devs to modify the code in future.

File: packages/core/src/hooks/auth/useLogin/index.spec.ts [hunk 3]
```diff
@@ -710,6 +710,47 @@ describe("useLogin Hook", () => {
...
+            expect(openNotificationMock).toBeCalledWith({
+                key: "login-success",
+                type: "success",
+                message: "Login successful",
+                description: "You are now logged in",
+            });
+        });
+    });
+    
 });
```
Comment (arghyapolley) [inline packages/core/src/hooks/auth/useLogin/index.spec.ts:754]: after each test case, a proper clean up will add efficiency, avoid false results, prevent resource leaks, and interference can also be avoided.

**Review by BatuhanW — APPROVED** [review]
Great work 🏅

File: .changeset/twelve-laws-grab.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+"@refinedev/core": minor
+---
+
+Updated the core and added successNotification field to AuthActionResponse and Updated relevant hooks to show success notification when successNotification object is provided and added specs.
```
Comment (omeraplak) [inline .changeset/twelve-laws-grab.md:5]:
```suggestion
---
"@refinedev/core": minor
---
feat(core): add success notification support for auth provider methods #5473

Updated the core and added successNotification field to AuthActionResponse and Updated relevant hooks to show success notification when successNotification object is provided and added specs.
```

File: packages/core/src/hooks/auth/useUpdatePassword/index.ts [hunk 5]
```diff
@@ -252,3 +257,14 @@ const buildNotification = (
         type: "error",
     };
 };
+
+const buildSuccessNotification = (
+    successNotification: SuccessNotificationResponse,
+  ): OpenNotificationParams => {
+    return {
+      message: successNotification.message || "Success",
```
Comment (alicanerdurmaz) [inline packages/core/src/hooks/auth/useUpdatePassword/index.ts:265]: message and descriptions are required but have fallback value. It confuses me. should we delete fallback values? what do you think? @Conqxeror

**Review by alicanerdurmaz — CHANGES_REQUESTED** [review]

**Review by alicanerdurmaz — APPROVED** [review]

**Pull request merged**